### DISCUSSION TITLE
Depth writing enabled/disabled per fragment

### DISCUSSION THREAD

**Discussion by omri-polycam** [google/filament#7392] [General]
Hi guys, is it possible to control whether you do or don't write to the z buffer within the fragment shader (something like glDepthMask())
I would like to use this when rendering a quadrilateral (as 2 triangles of course) that has a fragment shader with a high alpha near the center of the quad that goes to 0 near the edges. When this fragment shader calculated alpha is above a certain threshold I would like to write to the depth and color buffers as usual. When the alpha is below this threshold, I would like to still write to the color buffer, but not modify the depth buffer. This would be unrelated to depth testing, which is not enabled in this application.

Thanks,
   Omri

**Comment by pixelflinger** [comment]
No you cannot do that. The only thing you can do is call `discard` but the whole fragment will be aborted. Maybe you can use `gl_FragDepth`, you would write the depth, but with a value of your choice.

note that using `discard` or `gl_FragDepth` will have a significant performance impact.

**Reply by omri-polycam** [reply]
Thanks pixelflinger, I figured this was a long shot. Yeah, I'm aware of `discard` and `gl_FragDepth`, but neither approach would accomplish what I'm trying to do without unwanted side effects. I'm contemplating the idea of a 2nd pass that only writes to the depth buffer and not the color buffer and then I can simply use `discard` to control not writing to the depth buffer. However, if `discard` is very slow, that might not be a good way to go either.
